Sources/SwiftToolchain/ToolchainManagement.swift

```diff
@@ -70,16 +70,9 @@ extension FileSystem {
       }
     }
 
-    guard let cwd = currentWorkingDirectory else { return defaultToolchainVersion }
-
-    let versionFile = cwd.appending(component: ".swift-version")
-
-    guard isFile(versionFile), let version = try readFileContents(versionFile)
-      .validDescription?
-      // get the first line of the file
-      .components(separatedBy: CharacterSet.newlines).first,
-      version.contains("wasm")
-    else { return defaultToolchainVersion }
+    guard let version = try fetchLocalSwiftVersion() else {
```

Comment:
If the function is updated as in the suggestion below, the substring check would make sense here:

```suggestion
    guard 
      let version = try fetchLocalSwiftVersion(), 
      version.contains("wasm") 
    else {
```